models/gif_psc_exp.cpp

```diff
@@ -294,26 +290,21 @@ nest::gif_psc_exp::calibrate()
   // since t_ref_ >= 0, this can only fail in error
   assert( V_.RefractoryCounts_ >= 0 );
 
-  // initializing internal state
-  if ( !S_.sfa_stc_initialized_ )
-  {
-    V_.P_sfa_.clear();
-    V_.P_stc_.clear();
-
-    for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
-    {
-      V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
-    }
-    S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
+  // initializing adaptation (stc/sfa) variables
+  V_.P_sfa_.clear();
+  V_.P_stc_.clear();
 
-    for ( size_t i = 0; i < P_.tau_stc_.size(); i++ )
-    {
-      V_.P_stc_.push_back( std::exp( -h / P_.tau_stc_[ i ] ) );
-    }
-    S_.stc_elems_.resize( P_.tau_stc_.size(), 0.0 );
+  for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
+  {
+    V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
```

Comment:
It would be slightly faster and probably lead to a cleaner memory layout if you replaced the `clear()` above by `resize()` and then filled the elements here instead of using `push_back()`, which may require reallocation of the vector. Or you could follow the `clear()` with `reserve()` to make sure enough space is available.